src/OmniSharp.Host/Services/AssemblyLoader.cs

```diff
@@ -30,5 +33,43 @@ public Assembly Load(AssemblyName name)
             _logger.LogTrace($"Assembly loaded: {name}");
             return result;
         }
+
+        public IEnumerable<Assembly> LoadAll(string folderPath)
```

Comment:
Seems like we could return a better interface than `IEnumerable<Assembly>` since the instance is a `List<Assembly>`. Maybe `IReadOnlyList<Assembly>` so that callers could index into it if they want.